addons/addon-base-raas/packages/base-raas-services/lib/aws-accounts/aws-cfn-service.js

```diff
@@ -0,0 +1,203 @@
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const _ = require('lodash');
+const Service = require('@aws-ee/base-services-container/lib/service');
+// const { runAndCatch } = require('@aws-ee/base-services/lib/helpers/utils');
+const { allowIfActive, allowIfAdmin } = require('@aws-ee/base-services/lib/authorization/authorization-utils');
+const { processInBatches } = require('@aws-ee/base-services/lib/helpers/utils');
+
+// const { generateId } = require('../helpers/utils');
+
+/**
+ * This service is responsible for managing CFN stacks that provision AWS account permissions
+ */
+
+const settingKeys = {
+  awsRegion: 'awsRegion',
+};
+
+class AwsCfnService extends Service {
+  constructor() {
+    super();
+    this.boom.extend(['notSupported', 400]);
+    this.dependency([
+      'aws',
+      'jsonSchemaValidationService',
+      'authorizationService',
+      'auditWriterService',
+      'cfnTemplateService',
+      'awsAccountsService',
+    ]);
+  }
+
+  async init() {
+    await super.init();
+  }
+
+  /**
+   * Queries the stack at the data source AWS account and returns the following object:
+   * { stackId, templateId, templateVer, at }
+   *
+   * An exception is thrown if an error occurs while trying to describe the stack. This could happen if the stack
+   * is not created yet or is not provisioned in the correct account and region or was provisioned but did not
+   * use the correct stack name.
+   *
+   * @param requestContext
+   * @param accountEntity
+   */
+  async getStackTemplate(requestContext, accountEntity) {
+    await this.assertAuthorized(
+      requestContext,
+      { action: 'query-aws-cfn-stack', conditions: [allowIfActive, allowIfAdmin] },
+      { accountEntity },
+    );
+    const region = this.settings.get(settingKeys.awsRegion);
+    const { xAccEnvMgmtRoleArn, cfnStackName, externalId } = accountEntity;
+    const cfnApi = await this.getCfnSdk(xAccEnvMgmtRoleArn, externalId, region);
+    const params = { StackName: cfnStackName };
+    const stacks = await cfnApi.describeStacks(params).promise();
+    const stack = _.find(_.get(stacks, 'Stacks', []), item => item.StackName === cfnStackName);
+
+    if (_.isEmpty(stack)) {
+      throw this.boom.notFound(`Stack '${cfnStackName}' not found`, true);
+    }
+
+    const permissionsTemplateRaw = await cfnApi.getTemplate(params).promise();
+
+    return {
+      permissionsTemplateStr: permissionsTemplateRaw.TemplateBody,
```

Comment:
I guess not presently, my thought was that if we needed to query additional information later on it could be better to have it wrapped in an object because we could just add additional entries. Now that I think about it, I think if we're pulling other information that should probably be its own method, so I'll remove the object wrapping here and have it just remove the string.